lessons/guesser.rb

```diff
@@ -0,0 +1,77 @@
+def valid_answer?(input)
+    valid = ["too low", "too high", "correct"]
+    valid.include?(input.downcase)
+end
+
+def valid_yes_no?(input)
+    input.downcase == "yes" || input.downcase == "no"
+end
+
+def guessing_game
+    loop do
+        puts "Please think of a number from 1 to 100"
+        
+        low = 1
+        high = 100
+        guess = nil
+        lying = false
+
+        loop do 
+            #binary search
+            guess = (low + high) / 2
+            puts "Your number is #{guess}. Enter 'correct', too low', or 'too high'"
+            answer = gets.chomp.downcase
+
+            unless valid_answer?(answer)
+                puts "Invalid input. Enter 'correct', too low', or 'too high'"
+                next
+            end
+
+            if answer == "correct"
+                break
+            elsif answer == 'too low'
+                low = guess + 1
+            elsif answer == 'too high' 
+                high = guess - 1
+            end
+
+            #not sure if I need this if statement. looks the code will never get to this point
+            if high < low
+                lying = true
+                break
+            end
+        end
+
+        if high == low
+            guess = high
+            puts "Yay! I won! Your number is #{guess}, right?"
+
+            loop do
+                correct_answer = gets.chomp.downcase
+                break if valid_yes_no?(correct_answer)
+                puts "Invalid input. Please enter 'yes' or 'no'"
+            end
+
+            if correct_answer != 'yes'
+                puts "You lied! My guess is correct!"
+            end
```

Comment:
In this case, it could have been that the user lied about your guess being too high or too low at another point in the game. That is, there is no way to know which user input was a lie, and therefore we can only know that there is some inconsistency. 

This is obviously still fine, but I'm just being pedantic about how this code doesn't know for sure that its guess is correct.

Reply:
Maya, I agree with your point. We can't tell when the user lied about the number. We can only make that conclusion when we get to the last possible number. The code can know that its guess was correct just based on the received user's answers